system/security/plugins/jwtSecurity/jwtCache.cpp

```diff
@@ -0,0 +1,392 @@
+
+    return *this;
+}
+
+JWTUserInfo& JWTUserInfo::clearFeaturePerms()
+{
+    featurePermMap.clear();
+
+    return *this;
+}
+
+unsigned int JWTUserInfo::countFeaturePerms() const
+{
+    return featurePermMap.size();
+}
+
+//-----------------------------------------------------------------------------
+
+JWTUserInfo& JWTUserInfo::addWUScopePerm(const std::string& scope, SecAccessFlags accessFlag, SecAccessFlags defaultFlag, bool asDeny)
+{
+    return _addScopePerm(wuScopePermMap, scope, accessFlag, defaultFlag, asDeny);
+}
+
+JWTUserInfo& JWTUserInfo::addWUScopePerm(const std::string& scope, const std::string& accessFlagName, SecAccessFlags defaultFlag, bool asDeny)
+{
+    return _addScopePerm(wuScopePermMap, scope, getSecAccessFlagValue(accessFlagName.c_str()), defaultFlag, asDeny);
+}
+
+SecAccessFlags JWTUserInfo::matchWUScopePerm(const std::string& scope, SecAccessFlags defaultFlag) const
+{
+    return _matchScopePerm(wuScopePermMap, scope, defaultFlag);
+}
+
+JWTUserInfo& JWTUserInfo::eraseWUScopePerm(const std::string& scope)
+{
+    return _eraseScopePerm(wuScopePermMap, scope);
+}
+
+JWTUserInfo& JWTUserInfo::clearWUScopePerm()
+{
+    return _clearScopePerm(wuScopePermMap);
+}
+
+unsigned int JWTUserInfo::countWUScopePerms() const
+{
+    return _countScopePerms(wuScopePermMap);
+}
+
+bool JWTUserInfo::hasWUScopePerms() const
+{
+    return !wuScopePermMap.empty();
+}
+
+//-----------------------------------------------------------------------------
+
+JWTUserInfo& JWTUserInfo::addFileScopePerm(const std::string& scope, SecAccessFlags accessFlag, SecAccessFlags defaultFlag, bool asDeny)
+{
+    return _addScopePerm(fileScopePermMap, scope, accessFlag, defaultFlag, asDeny);
+}
+
+JWTUserInfo& JWTUserInfo::addFileScopePerm(const std::string& scope, const std::string& accessFlagName, SecAccessFlags defaultFlag, bool asDeny)
+{
+    return _addScopePerm(fileScopePermMap, scope, getSecAccessFlagValue(accessFlagName.c_str()), defaultFlag, asDeny);
+}
+
+SecAccessFlags JWTUserInfo::matchFileScopePerm(const std::string& scope, SecAccessFlags defaultFlag) const
+{
+    return _matchScopePerm(fileScopePermMap, scope, defaultFlag);
+}
+
+JWTUserInfo& JWTUserInfo::eraseFileScopePerm(const std::string& scope)
+{
+    return _eraseScopePerm(fileScopePermMap, scope);
+}
+
+JWTUserInfo& JWTUserInfo::clearFileScopePerm()
+{
+    return _clearScopePerm(fileScopePermMap);
+}
+
+unsigned int JWTUserInfo::countFileScopePerms() const
+{
+    return _countScopePerms(fileScopePermMap);
+}
+
+bool JWTUserInfo::hasFileScopePerms() const
+{
+    return !fileScopePermMap.empty();
+}
+
+//-----------------------------------------------------------------------------
+// Private methods
+//-----------------------------------------------------------------------------
+
+std::string JWTUserInfo::_globToExample(const std::string& pattern) const
+{
+    std::string     s1 = std::regex_replace(pattern, std::regex("\\[![0-9].+?\\]\\*?"), "a");
```

Comment:
Likely to be more efficient if std::regex were static constants outside the function.  I don't know how time critical this function is though.

Reply:
I've moved the std::regex parameters to static const std::regex file variables.